packages/dapp-high-roller/src/components/app-game/app-game.tsx

```diff
@@ -134,30 +135,26 @@ export class AppGame {
       console.log("Reset appInstance", this.appInstance);
     }
 
-    if (this.highRollerState.stage === HighRollerStage.PRE_GAME) {
+    if (
+      this.highRollerState.stage === HighRollerStage.WAITING_FOR_P1_COMMITMENT
+    ) {
       await this.beginRolling("myRoll");
 
-      const startGameAction: Action = {
-        number: 0,
-        actionType: ActionType.START_GAME,
-        actionHash: HashZero
-      };
-
-      this.highRollerState = (await this.appInstance.takeAction(
-        startGameAction
-      )) as HighRollerAppState;
-
       const numberSalt = generateSalt();
+
       const playerFirstNumber =
         1 + Math.floor(Math.random() * Math.floor(1000));
+
       const hash = computeCommitHash(numberSalt, playerFirstNumber);
 
-      const commitHashAction: Action = {
-        number: 0,
-        actionType: ActionType.COMMIT_TO_HASH,
-        actionHash: hash
+      const commitHashAction: HighRollerAction = {
+        actionType: HighRollerActionType.COMMIT_TO_HASH,
+        actionHash: hash,
+        // NOTE: `number` is unused with COMMIT_TO_HASH but cannot be undefined
+        number: 0
       };
 
+      // TODO: Figure out why this is using ...
```

Comment:
This is just spreading out the response from `takeAction` so the properties on the `HighRollerAppState` object get applied as is to `this.highRollerState`